### PR TITLE
Set up basic Battle page scaffolding

### PR DESCRIPTION
**Pull request by dbmikus**
Set up the basic Battle page scaffolding. It just sets up the state machine but doesn't complete the implementation of any transitions based on events.

Along the way, we:

* Cleaned up Scaffold ETH routes and add empty Battle route
* Set up a skeleton `<Battle ... />` component. Added the states and the names of events that map between states
* Add and refactor code to work with homelands/realms on the frontend. Have a new `helpers/lands.js` file and a new `Land` component
* Move gameplay reducer stuff into its own hook at `hooks/gameplay`

### CODE REVIEW

**Review by bensadeghi — APPROVED** [review]
Awesome!!

But one change is needed either on the front or back-end:
Last night I changed the contract to save the homeland/realm data as a `string`, which is randomly selected from ["Earth", "Ice", "Fire", "Water", "Stars", "Sky"]. This made the svg rendering and populating the metadata really easy.
https://github.com/dbmikus/ethonline-2021-nft-cards/blob/master/packages/hardhat/contracts/BlockMonsters.sol#L164

I can change the field back to an integer in the contract, or we can change the front-end accordingly

File: packages/react-app/src/views/Battle.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  useGameplayReducer,
+  PRE_START,
+  DECK_BUILDING,
+  MONSTER_SELECTION,
+  MONSTER_BATTLEFIELD,
+  GAME_END,
+} from "../hooks/gameplay";
+
+/**
+ * Components:
```
Comment (dbmikus) [inline packages/react-app/src/views/Battle.jsx:14]: @bensadeghi this is the page/component that can connect with the deck builder you are working on.

**Comment by dbmikus** [comment]
I'll change the frontend.

**Comment by dbmikus** [comment]
Frontend fixed to use realm strings instead of numbers. I'm going to merge this now before I go to sleep. Thanks for the quick review.

**Pull request merged**